master/buildbot/steps/source/git.py

```diff
@@ -752,6 +758,17 @@ def _doCommit(self):
         for message in self.messages:
             cmd.extend(['-m', message])
 
+        if self.emptyCommits == 'create-empty-commit':
+            cmd.extend(['--allow-empty'])
+
+        if self.emptyCommits == 'ignore':
+            stdout, rc = yield self._dovccmd(cmd, collectStdoutAndRC=True, decodeRC={0: SUCCESS, 1: SUCCESS})
+            if rc == 0:
+                return 0
+            if rc == 1 and ("nothing added to commit" in stdout or "nothing to commit" in stdout):
```

Comment:
what if lang is french? I would prefer to test  first with a git status --porcelain

Reply:
Hmmm. Bad luck? :) I've tried to implement this in 3e46d7814c8aa73a775a67d4b2460cbe7f142fa2.